gradysim/protocol/addons/statistics.py

```diff
@@ -0,0 +1,209 @@
+"""
+This module contains a function that creates statistics which wraps a protocol instance and it's methods. Implements
+a call chain for each of the protocol interface's methods.
+
+Use this module through the **create_statistics**][gradysim.protocol.addons.statistics.create_statistics] method,
+**never** instantiate the StatisticsProtocolWrapper directly.
+
+Beware that this module uses monkey patching and may result in broken protocols if someone else tries to tamper with
+the protocol's methods.
+"""
+
+import time
+from typing import Any, Dict, List
+
+import pandas as pd
+from gradysim.protocol.addons.dispatcher import create_dispatcher, DispatchReturn
+
+from gradysim.protocol.interface import IProtocol
+from gradysim.protocol.messages.telemetry import Telemetry
+
+# Wrapper methods for handler
+def initialize_srt(protocol: IProtocol) -> DispatchReturn:
+    _statistics_protocol_wrappers[protocol].update_srt_statistic(
+        protocol.provider.current_time(), time.time()
+    )
+    return DispatchReturn.CONTINUE
+
+
+def handle_timer_srt(protocol: IProtocol, timer: str) -> DispatchReturn:
+    _statistics_protocol_wrappers[protocol].update_srt_statistic(
+        protocol.provider.current_time(), time.time()
+    )
+    return DispatchReturn.CONTINUE
+
+
+def handle_packet_srt(protocol: IProtocol, message: str) -> DispatchReturn:
+    _statistics_protocol_wrappers[protocol].update_srt_statistic(
+        protocol.provider.current_time(), time.time()
+    )
+    return DispatchReturn.CONTINUE
+
+
+def handle_telemetry_srt(protocol: IProtocol, telemetry: Telemetry) -> DispatchReturn:
+    _statistics_protocol_wrappers[protocol].update_srt_statistic(
+        protocol.provider.current_time(), time.time()
+    )
+    return DispatchReturn.CONTINUE
+
+
+def finish_srt(protocol: IProtocol) -> DispatchReturn:
+    _statistics_protocol_wrappers[protocol].update_srt_statistic(
+        protocol.provider.current_time(), time.time()
+    )
+    return DispatchReturn.CONTINUE
+
+
+def handle_packet_tv(protocol: IProtocol, message: str) -> DispatchReturn:
+    _statistics_protocol_wrappers[protocol].update_tracked_variable_statistic(
+        protocol.provider.current_time(), protocol.provider.tracked_variables
+    )
+    return DispatchReturn.CONTINUE
```

Comment:
I think these are all private right? Add a "_" to the start of the method so it doesn't show up in autocomplete when importing or in the docs